modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java

```diff
@@ -390,8 +390,19 @@ public void revert(long previousLevel) {
         // walk back removing blocks greater than the given level value
         IAionBlock bestLine = bestBlock;
         while (currentLevel > previousLevel) {
+
+            // remove all the blocks at that level
+            List<BlockInfo> currentLevelBlocks = getBlockInfoForLevel(currentLevel);
+            if (currentLevelBlocks.size() == 0){
```

Comment:
WIll this ever be true? given that we just grabbed the latest block from blockInfo

Reply:
since the BlockInfo for the block can be null, I think it is possible:
https://github.com/aionnetwork/aion/blob/35a24fbbeef1e74b9d7068ab12b8bf99f556d952/modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java#L327-L332

but it's also possible that the `currentLevelBlocks` is null, so I'll add a null check

Reply:
LGTM